src/traces/scatter/select.js

```diff
@@ -40,6 +41,11 @@ module.exports = function selectPoints(searchInfo, polygon) {
             di = cd[i];
             x = xa.c2p(di.x);
             y = ya.c2p(di.y);
+
+            if(x === BADNUM || y === BADNUM) {
+                continue;
+            }
```

Comment:
Good call. Done in https://github.com/plotly/plotly.js/pull/1656/commits/80cee013a0a6b70092fc1ecb3ac7ba6ed812d81d